### PR TITLE
feat: support automatically translated messages

### PR DESCRIPTION
**Pull request by madsroskar**
This PR is to be treated as a draft/not complete PR at the moment. The functionality works, but I've spent my last two brain cells trying to figure out all the different intentions of the code, so my refactoring is mostly just me parsing it piece by piece after writing some tests. 

To be completed.

### CODE REVIEW

File: package/src/components/ChannelPreview/hooks/useLatestMessagePreview.ts [hunk 1]
```diff
@@ -29,74 +30,112 @@ export type LatestMessagePreview<
...
+>(
+  message: LatestMessage<StreamChatGenerics>,
+) => {
+  const shortenedText = truncateMessageText(message) || '';
+
+  const mentionedUsers = message.mentioned_users
+    ?.map((user: User) => `(@${user.name || user.id || ''})`)
+    .join('|');
+
+  const mentionedUsersRegExp = new RegExp(`${mentionedUsers}|(!:${mentionedUsers})`, 'mg');
```
Comment (madsroskar) [inline package/src/components/ChannelPreview/hooks/useLatestMessagePreview.ts:83]: Note to self: the or here is unnecessary - can just be `mentionedUsers`

**Review by vanGalilea — CHANGES_REQUESTED** [review]

File: examples/SampleApp/src/hooks/useChatClient.ts [hunk 2]
```diff
@@ -79,6 +79,7 @@ export const useChatClient = () => {
     const user = {
       id: config.userId,
       image: config.userImage,
+      language: 'es' as TranslationLanguages,
```
[the remark this answers is not in the document]
Comment (madsroskar) [reply examples/SampleApp/src/hooks/useChatClient.ts:82]: We have a note somewhere to set up testing stuff, we could create a dev screen for things like this. I'll remove it from this PR though

[the remark this answers is not in the document]
Comment (vanGalilea) [reply examples/SampleApp/src/hooks/useChatClient.ts:82]: I'll create an issue for this, render and copy channel id, message-id

**Comment by madsroskar** [comment]
Closing this, creating a new PR

**Pull request closed without merging**